src/ConversationCursor.ts

```diff
@@ -69,28 +69,45 @@ export class ConversationCursor {
         lastReadMessageTimestampFetch: (roomId: string) => BasicMessageInfo | undefined,
         options?: CursorOptions
     ) {
-        const limit = ConversationCursor.calculateLimit(options)
-        const initialSize = options?.initialSize ?? this.DEFAULT_INITIAL_SIZE
-        const room = client.getRoom(roomId)!
-        const timelineSet = getOnlyMessagesTimelineSetFromRoom(userId, room, limit)
-        const window = new TimelineWindow(client, timelineSet, { windowLimit: limit })
-        await window.load(initialEventId, initialSize)
-
-        // It could happen that the initial size of the window isn't respected. That's why we will try to fix it
-        let windowSize = window.getEvents().length
-        let gotResults = true
-        while (windowSize < initialSize && gotResults) {
-            gotResults = await window.paginate(EventTimeline.BACKWARDS, initialSize - windowSize)
-            windowSize = window.getEvents().length
-        }
+        try {
+            const limit = ConversationCursor.calculateLimit(options)
+            const initialSize = options?.initialSize ?? this.DEFAULT_INITIAL_SIZE
+            let room = client.getRoom(roomId)
 
-        gotResults = true
-        while (windowSize < initialSize && gotResults) {
-            gotResults = await window.paginate(EventTimeline.FORWARDS, initialSize - windowSize)
-            windowSize = window.getEvents().length
-        }
+            for (let i = 0; i < 3; i++) {
+                if (!room) {
+                    await delay(1000)
```

Comment:
what do you say if we increase this delay or retries so we at least wait for a sync call which has 30s of timeout?